### PR TITLE
reject requests to publish projects from users under 13

### PR DESCRIPTION
**Pull request by davidsbailey**
This PR makes it so that projects belonging to users under age 13 cannot be published. The work to hide the "Publish" buttons in the UI will be done in a separate PR by @caleybrock . The publish/unpublish buttons are currently hidden behind an experiment flag, so there is no user impact to launching the server side of this before the buttons are hidden on the client.

There is some discussion around the stubbing approach here: https://codedotorg.slack.com/archives/C0T10HG6N/p1495549227952810

### CODE REVIEW

File: shared/middleware/helpers/auth_helpers.rb [hunk 1]
```diff
@@ -18,6 +18,14 @@ def current_user
   @dashboard_user ||= DASHBOARD_DB[:users][id: current_user_id]
 end
 
+# Returns true if the current user is under 13 or if age is unknown.
+def under_13?
+  return true unless current_user
+  birthday = current_user[:birthday]
+  age = ((Date.today - birthday) / 365).to_i
+  age < 13
+end
```
Comment (ashercodeorg) [inline shared/middleware/helpers/auth_helpers.rb]: This seems to be duplicating [`User#under_13?`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L769-L771) in a sense. Is it possible to share the code? If not, I'd recommend adding a comment in both places indicating that the code is duplicated.

Comment (davidsbailey) [reply shared/middleware/helpers/auth_helpers.rb]: How specifically would you recommend that the code be shared? `User#under_13?` uses the rails model, which we don't have access to here.

Comment (davidsbailey) [reply shared/middleware/helpers/auth_helpers.rb]: To answer your question, the only line I see a way to easily share is `age = ((Date.today - birthday) / 365).to_i`, and I don't think that's enough complexity to warrant sharing. I'd be happy to add a comment explaining that the logic is duplicated.

Comment (ashercodeorg) [reply shared/middleware/helpers/auth_helpers.rb]: Not sure there is a great way, thus my asking questions. Perhaps in [user_helpers.rb](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/user_helpers.rb)?

Comment (davidsbailey) [reply shared/middleware/helpers/auth_helpers.rb]: If you insist :) PTAL

**Comment by ashercodeorg** [comment]
LGTM.

**Pull request merged**